options/posix/include/mlibc/lookup.hpp

```diff
@@ -18,6 +18,13 @@ struct dns_addr_buf {
 	uint8_t addr[16];
 };
 
+struct lookup_result {
+	lookup_result() : buf(getAllocator()), aliases(getAllocator())
+	{}
```

Comment:
Coding style wise:
```suggestion
	lookup_result()
	: buf(getAllocator()), aliases(getAllocator()) {}
```
Also, `{}` initialization might be preferable to `()` initialization in general.